src/v/cloud_topics/housekeeper/housekeeper.cc

```diff
@@ -58,6 +58,9 @@ ss::future<> housekeeper::do_housekeeping() {
     if (new_start_offset != kafka::offset::min()) {
         co_await _l0_metastore->set_start_offset(_tidp, new_start_offset, &_as);
     }
+    // Sync the start offset back to the L1 metastore.
+    // DeleteRecords may advance the L0 start offset past the L1 start offset.
+    co_await sync_start_offset();
```

Comment:
Not necessarily related to this PR, but if retention computes the same offset, we should catch that and skip both of these calls..

Though maybe we'd want to cache the last successful L1 start offset in case a previous sync call had failed and we need to retry